vllm/v1/engine/core.py

```diff
@@ -143,17 +150,36 @@ def abort_requests(self, request_ids: List[str]):
         self.scheduler.finish_requests(request_ids,
                                        RequestStatus.FINISHED_ABORTED)
 
+        self.guided_decoding_manager.remove_requests(request_ids)
+
     def step(self) -> EngineCoreOutputs:
         """Schedule, execute, and make output."""
 
         if not self.scheduler.has_unfinished_requests():
             return EngineCoreOutputs(
                 outputs=[], scheduler_stats=self.scheduler.make_stats())
 
+        # Check for cached grammars and allocate bitmask if necessary
+        self.setup_grammars()
```

Comment:
After this change we can remove `setup_grammars()` from this class since it's just a direct passthrough.

```suggestion
        self.guided_decoding_manager.setup_grammars()
```